chips/msp432/src/cs.rs

```diff
@@ -0,0 +1,275 @@
+    CSKEY [
+        // for accessing any other register, it must be unlocked using this key-register
+        KEY OFFSET(0) NUMBITS(16)
+    ],
+    CSCTL0 [
+        // for calibrating the DCO frequency
+        DCOTUNE OFFSET(0) NUMBITS(10),
+        // DCO frequency range select
+        DCORSEL OFFSET(16) NUMBITS(3),
+        // enable/disable DCO external resistor mode
+        DCORES OFFSET(22) NUMBITS(1),
+        // enable DCO
+        DCOEN OFFSET(23) NUMBITS(23)
+    ],
+    CSCTL1 [
+        // select MCLK source
+        SELM OFFSET(0) NUMBITS(3),
+        // select SMCLK and HSMCLK source
+        SELS OFFSET(4) NUMBITS(3),
+        // selects ACLK source
+        SELA OFFSET(8) NUMBITS(3),
+        // selects BLCK source
+        SELB OFFSET(12) NUMBITS(1),
+        // MCLK source divider
+        DIVM OFFSET(16) NUMBITS(3),
+        // HSMCLK source divider
+        DIVHS OFFSET(20) NUMBITS(3),
+        // ACLK source divider
+        DIVA OFFSET(24) NUMBITS(3),
+        // SMCLK divider
+        DIVS OFFSET(28) NUMBITS(3)
+    ],
+    CSCTL2 [
+        // set drive-strength for LXFT oscillator
+        LFXTDRIVE OFFSET(0) NUMBITS(2),
+        // turn on LFXT oscillator
+        LFXT_EN OFFSET(8) NUMBITS(1),
+        // LFXT bypass select
+        LFXTBYPASS OFFSET(9) NUMBITS(1),
+        // HFXT oscillator drive selection
+        HFXTDRIVE OFFSET(16) NUMBITS(1),
+        // HFXT frequency selection
+        HFXTFREQ OFFSET(20) NUMBITS(3),
+        // turn on HFXT oscillator
+        HFXT_EN OFFSET(24) NUMBITS(1),
+        // HFXT bypass select
+        HFXTBYPASS OFFSET(25) NUMBITS(1)
+    ],
+    CSCTL3 [
+        // start flag counter for LFXT
+        FCNTLF OFFSET(0) NUMBITS(2),
+        // reset start fault counter for LFXT
+        RFCNTLF OFFSET(2) NUMBITS(1),
+        // enable start fault counter for LFXT
+        FCNTLF_EN OFFSET(0) NUMBITS(1),
+        // start flag counter for HFXT
+        FCNTHF OFFSET(4) NUMBITS(2),
+        // reset start fault counter for HFXT
+        RFCNTHF OFFSET(6) NUMBITS(1),
+        // enable start fault counter for HFXT
+        FCNTHF_EN OFFSET(7) NUMBITS(1)
+    ],
+    CSCLKEN [
+        // ACLK system clock conditional request enable
+        ACLK_EN OFFSET(0) NUMBITS(1),
+        // MCLK system clock conditional request enable
+        MCLK_EN OFFSET(1) NUMBITS(1),
+        // HSMCLK system clock conditional request enable
+        HSMCLK_EN OFFSET(2) NUMBITS(1),
+        // SMCLK system clock conditional request enable
+        SMCLK_EN OFFSET(3) NUMBITS(1),
+        // turn on the VLO oscillator
+        VLO_EN OFFSET(8) NUMBITS(1),
+        // turn on the REFO oscillator
+        REFO_EN OFFSET(9) NUMBITS(1),
+        // turn on the MODOSC oscillator
+        MODOSC_EN OFFSET(10) NUMBITS(1),
+        // select REFO nominal frequency: 0 = 32.768kHz, 1=128kHz
+        REFOFSEL OFFSET(15) NUMBITS(1)
+    ],
+    // status of the different clock-sources, if they are active or not
```

Comment:
Please promote these to doc comments (`//` -> `///`) and add descriptors for the other register types as well so that documentation is generated for them.

Reply:
Of course, I oversaw these comments.